src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -89,6 +91,15 @@ static bool exclusive = false;
 #endif
 #define htonll(a) ntohll(a)
 
+static int do_unmap();
+
+static void handle_signal(int signum)
+{
+  assert(signum == SIGINT || signum == SIGTERM);
+  derr << "*** Got signal " << sig_str(signum) << " ***" << dendl;
+  ioctl(nbd, NBD_DISCONNECT);
```

Comment:
Testing the only issue noticed is that if there are many pending I/O requests, the NBD_DISCONNECT ioctl may last long time (the same is when doing unmap via CLI). Thus it it would be nice to add some dignostic: "sending NBD_DISCONNECT" debug message before ioctl, check for return value and output error if it failed (a debug message if it succeeded).